pkg/bindinfo/binding_cache.go

```diff
@@ -448,11 +390,28 @@ func (c *bindingCache) GetAllBindings() Bindings {
 
 // SetBinding sets the Bindings to the cache.
 // The function is thread-safe.
-func (c *bindingCache) SetBinding(sqlDigest string, meta Bindings) (err error) {
+func (c *bindingCache) SetBinding(sqlDigest string, bindings Bindings) (err error) {
 	c.lock.Lock()
 	defer c.lock.Unlock()
+
+	// prepare noDBDigests for all bindings
+	noDBDigests := make([]string, 0, len(bindings))
+	p := parser.New()
+	for _, binding := range bindings {
+		stmt, err := p.ParseOneStmt(binding.BindSQL, binding.Charset, binding.Collation)
+		if err != nil {
+			return err
+		}
+		_, noDBDigest := norm.NormalizeStmtForBinding(stmt, norm.WithoutDB(true))
+		noDBDigests = append(noDBDigests, noDBDigest)
+	}
+
+	for i, binding := range bindings {
+		c.digestBiMap.Add(noDBDigests[i], binding.SQLDigest)
+	}
+
 	cacheKey := bindingCacheKey(sqlDigest)
-	_, err = c.set(cacheKey, meta)
+	_, err = c.set(cacheKey, bindings)
```

Comment:
It seems we dropped the original comment here.

Reply:
Updated. Thanks.